### PR TITLE
feat(build): refactor cypress tests, add some tests for invite users, domain creation

### PR DESCRIPTION
**Pull request by anshbansal**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: smoke-test/tests/cypress/cypress/integration/glossary/glossary.js [hunk 1]
```diff
@@ -2,24 +2,25 @@ describe("glossary", () => {
     it("go to glossary page, create terms, term group", () => {
 
         const urn = "urn:li:dataset:(urn:li:dataPlatform:hive,cypress_logging_events,PROD)";
+        const dataset_name = "cypress_logging_events";
```
Comment (gabe-lyons) [inline smoke-test/tests/cypress/cypress/integration/glossary/glossary.js:5]: nit: js vars are always `lowerCamelCase`

File: smoke-test/tests/cypress/cypress/integration/lineage/impact_analysis.js [hunk 2]
```diff
@@ -1,58 +1,45 @@
...
 
-    // impact analysis can take a beat- don't want to time out here
-    cy.wait(5000);
+describe("impact analysis", () => {
+  it("can see 1 hop of lineage by default", () => {
+    startAtDataSetLineage()
 
-    cy.contains("User Creations").should("not.exist");
-    cy.contains("User Deletions").should("not.exist");
+    cy.ensureTextNotPresent("User Creations");
```
Comment (gabe-lyons) [inline smoke-test/tests/cypress/cypress/integration/lineage/impact_analysis.js:14]: oh nice!

File: smoke-test/tests/cypress/cypress/integration/lineage/impact_analysis.js [hunk 3]
```diff
@@ -1,58 +1,45 @@
...
-  it("can see 1 hop of lineage by default", () => {
+const startAtDataSetLineage = () => {
     cy.login();
-    cy.visit(
-      "/dataset/urn:li:dataset:(urn:li:dataPlatform:kafka,SampleCypressKafkaDataset,PROD)/Lineage?is_lineage_mode=false"
+    cy.goToDataset(
+      "urn:li:dataset:(urn:li:dataPlatform:kafka,SampleCypressKafkaDataset,PROD)",
+      "SampleCypressKafkaDataset"
     );
+    cy.openEntityTab("Lineage")
```
Comment (gabe-lyons) [inline smoke-test/tests/cypress/cypress/integration/lineage/impact_analysis.js:7]: ah this is so nice too

File: smoke-test/tests/cypress/cypress/integration/mutations/add_users.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            cy.visit("/signup?invite_token=bad_token");
+            tryToSignUp()
+            cy.waitTextVisible("Failed to log in! An unexpected error occurred.")
+        });
+    });
+});
+
+// Verify you can’t generate a reset password link for a non-native user (root, for example)
+// Generate a reset password link for a native user
+// Log out, then verify that using a bad reset token in the URL doesn’t allow you to reset password
```
Comment (gabe-lyons) [inline smoke-test/tests/cypress/cypress/integration/mutations/add_users.js:39]: lets move this comment to above the method its describing

File: smoke-test/tests/cypress/cypress/integration/mutations/deprecations.js [hunk 5]
```diff
@@ -1,14 +1,15 @@
 describe("deprecation", () => {
     it("go to dataset and check deprecation works", () => {
         const urn = "urn:li:dataset:(urn:li:dataPlatform:hive,cypress_logging_events,PROD)";
+        const dataset_name = "cypress_logging_events";
```
Comment (gabe-lyons) [inline smoke-test/tests/cypress/cypress/integration/mutations/deprecations.js:4]: nit: `lowerCamelCase`

File: smoke-test/tests/cypress/cypress/integration/mutations/domains.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+            .invoke('attr', 'data-testid')
+            .then((data_test_id) => {
+                cy.log(data_test_id)
+            })
+    })
+
+    // add asset to domain
+    // Search filter by domain
+    // Remove entity from domain
+    // Delete a domain - ensure that the dangling reference is deleted on the asset
```
Comment (gabe-lyons) [inline smoke-test/tests/cypress/cypress/integration/mutations/domains.js:24]: if these are to-do's lets mark them as todos?

File: smoke-test/tests/cypress/cypress/integration/mutations/mutations.js [hunk 7]
```diff
@@ -2,15 +2,14 @@ describe("mutations", () => {
   before(() => {
     // warm up elastic by issuing a `*` search
     cy.login();
-    cy.visit("http://localhost:9002/search?query=%2A");
```
Comment (gabe-lyons) [inline smoke-test/tests/cypress/cypress/integration/mutations/mutations.js:5]: thanks

File: smoke-test/tests/cypress/integration_test.py [hunk 8]
```diff
@@ -27,6 +27,7 @@ def test_run_cypress(frontend_session, wait_for_healthchecks):
         command = "NO_COLOR=1 npx cypress run --record"
     else:
         print('Running Cypress tests without recording')
+        # command = "NO_COLOR=1 npx cypress --version"
```
Comment (gabe-lyons) [inline smoke-test/tests/cypress/integration_test.py:30]: should we delete?

**Review by gabe-lyons — APPROVED** [review]
approving to unblock

File: smoke-test/tests/cypress/integration_test.py (hunk 8 above)
Comment (anshbansal) [reply smoke-test/tests/cypress/integration_test.py:30]: No. This should be kept as-it-is.

File: smoke-test/tests/cypress/cypress/integration/mutations/add_users.js (hunk 4 above)
Comment (anshbansal) [reply smoke-test/tests/cypress/cypress/integration/mutations/add_users.js:39]: these are todos. I'll automate these this week.

File: smoke-test/tests/cypress/cypress/integration/mutations/domains.js (hunk 6 above)
Comment (anshbansal) [reply smoke-test/tests/cypress/cypress/integration/mutations/domains.js:24]: these are todos. I'll automate these this week.

**Pull request merged**